WalletWasabi/Stores/IndexStore.cs

```diff
@@ -486,5 +495,31 @@ public async Task ForeachFiltersAsync(Func<FilterModel, Task> todo, Height fromH
 				}
 			}
 		}
+
+		private void ThrowIfDisposed()
+		{
+			if (_disposed)
+			{
+				throw new ObjectDisposedException(nameof(IndexStore));
+			}
+		}
+
+		public async ValueTask DisposeAsync()
+		{
+			if (_disposed)
+			{
+				return;
+			}
+
+			// Indicate that the object is disposed.
+			_disposed = true;
+
+			// Wait for the ongoing operations to finish, like TryCommitToFileAsync.
+			using (await MatureIndexAsyncLock.LockAsync().ConfigureAwait(false))
+			using (await ImmatureIndexAsyncLock.LockAsync().ConfigureAwait(false))
+			using (await IndexLock.LockAsync().ConfigureAwait(false))
+			{
+			}
```

Comment:
I would use that `using var _ = ...` syntax here too. With the same remark as before to explain why you do it.

Reply:
Can you elaborate on this? It is the same, isn't it?

Reply:
This is equivalent code:

```csharp
                        // Wait for the ongoing operations related to locks.
			using var _ = await IndexLock.LockAsync().ConfigureAwait(false);
			using var _ = await MatureIndexAsyncLock.LockAsync().ConfigureAwait(false);
			using var _ = await ImmatureIndexAsyncLock.LockAsync().ConfigureAwait(false);
```

and you don't need those braces,

Reply:
![image](https://user-images.githubusercontent.com/9844978/99405916-95c4e080-28ed-11eb-95d0-6e7acecac2b4.png)

Reply:
That's interesting. I thought that `_` is unnamed variable. Anyway, any name will do the job here.